concordium-consensus/src/Concordium/Skov/Update.hs

```diff
@@ -555,11 +556,29 @@ doReceiveTransactionInternal origin tr ts slot = do
           ptrs <- getPendingTransactions
           case wmdData of
             NormalTransaction tx -> do
-              putPendingTransactions $! addPendingTransaction (transactionNonce tx) WithMetadata{wmdData=tx,..} ptrs
+              -- Transactions received individually should always be added to ptt.
+              -- If the transaction was received as part of a block we only add it to ptt if
+              -- the transaction nonce is at least the `nextNonce` recorded for sender account.
```

Comment:
Technically this is describing what is happening below, which is not that useful. It would be more useful to explain **why**.